compiler/optimizer/LoopVersioner.cpp

```diff
@@ -3285,6 +3292,13 @@ bool TR_LoopVersioner::detectChecksToBeEliminated(TR_RegionStructure *whileLoop,
          _nullCheckReference = NULL;
          _inNullCheckReference = false;
 
+         // If the node is a GC point (either cause GC and return, or cause GC and throw),
+         // be conservative and do not version wrtbars. Applies to top level nodes/checks
+         // and child call nodes.
+         if (currentNode->canCauseGC() || ((currentNode->getNumChildren() > 0) &&
```

Comment:
Is this really necessary to check both the parent and the child node for canCauseGC ? 
Maybe this kind of checking on the child ought to be folded into the canCauseGC routine so that every caller does not have to do logic such as this.

Reply:
Upon inspecting `Node::canCauseGC` more closely, it does take into account both top level and child nodes, so calling it on the top level node is sufficient.

Reply:
Thanks, the code looks simpler and cleaner now in this area.